### PR TITLE
CLI: make snapshot name requiered in creating volume snapshots

### PR DESCRIPTION
**Pull request by yazdan**
- Make snapshot name argument requiered
- Update documentation and helps

### CODE REVIEW

File: command/volume_snapshot_create.go [hunk 1]
```diff
@@ -19,17 +19,16 @@ type VolumeSnapshotCreateCommand struct {
 
 func (c *VolumeSnapshotCreateCommand) Help() string {
 	helpText := `
-Usage: nomad volume snapshot create <volume id> [snapshot_name]
+Usage: nomad volume snapshot create <volume id> snapshot_name
```
Comment (jrasell) [inline command/volume_snapshot_create.go:22]: Let's keep the consistency seeing as this is now a required parameter:
```suggestion
Usage: nomad volume snapshot create <volume id> <snapshot_name>
```

File: command/volume_snapshot_create.go [hunk 2]
```diff
@@ -100,16 +99,13 @@ func (c *VolumeSnapshotCreateCommand) Run(args []string) int {
 
 	// Check that we at least one argument
 	args = flags.Args()
-	if l := len(args); l == 0 {
-		c.Ui.Error("This command takes at least one argument: <vol id> [snapshot name]")
+	if l := len(args); l <= 1 {
```
Comment (jrasell) [inline command/volume_snapshot_create.go:102]: I think this check is incorrect as we now want the argument length to be 2. We could therefore make this check `!= 2` or similar.

File: command/volume_snapshot_create.go [hunk 3]
```diff
@@ -100,16 +99,13 @@ func (c *VolumeSnapshotCreateCommand) Run(args []string) int {
...
+		c.Ui.Error("This command takes two argument: <vol id> <snapshot name>")
```
Comment (jrasell) [inline command/volume_snapshot_create.go:103]: Grammar nit:
```suggestion
		c.Ui.Error("This command takes two arguments: <vol id> <snapshot name>")
```

**Review by jrasell — CHANGES_REQUESTED** [review]
Hi @yazdan and thanks for raising this PR. I have added some inline comments and suggestions; please let me know if you have any follow up questions.

It would be good to add a changelog entry for this also. The `make cl` command can guide you through this.

File: command/volume_snapshot_create.go (hunk 1 above)
Comment (yazdan) [reply command/volume_snapshot_create.go:22]: done

File: command/volume_snapshot_create.go (hunk 3 above)
Comment (yazdan) [reply command/volume_snapshot_create.go:103]: both done

**Comment by yazdan** [comment]
@jrasell done, please review again.

File: .changelog/17958.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+cli: make snapshot name requiered in creating volume snapshots
```
Comment (jrasell) [inline .changelog/17958.txt:2]:
```suggestion
cli: Snapshot name is required in `volume snapshot create` command
```

**Review by jrasell — APPROVED** [review]
LGTM and thanks @yazdan! I have made a suggestion to the changelog entry, once this is updated I will get this merged and backported.

**Comment by yazdan** [comment]
@jrasell done!

**Pull request merged**